admin/server/users.go

```diff
@@ -163,6 +163,161 @@ func (s *Server) UpdateUserPreferences(ctx context.Context, req *adminv1.UpdateU
+		attribute.String("args.user_id", req.UserId),
+	)
+
+	claims := auth.GetClaims(ctx)
+	forceAccess := claims.Superuser(ctx) && req.SuperuserForceAccess
+
+	userID := req.UserId
+	if userID == "current" { // Special alias for the current user
+		if claims.OwnerType() != auth.OwnerTypeUser {
+			return nil, status.Error(codes.Unauthenticated, "not authenticated as a user")
+		}
+		userID = claims.OwnerID()
+	}
+	if userID != claims.OwnerID() && !forceAccess {
+		return nil, status.Error(codes.PermissionDenied, "not authorized to list auth tokens for other users")
+	}
+
+	pageSize := validPageSize(req.PageSize)
+	pageToken, err := unmarshalPageToken(req.PageToken)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	authTokens, err := s.admin.DB.FindUserAuthTokens(ctx, userID, pageToken.Val, pageSize)
+	if err != nil {
+		return nil, err
+	}
+
+	nextToken := ""
+	if len(authTokens) >= pageSize {
+		nextToken = marshalPageToken(authTokens[len(authTokens)-1].ID)
+	}
+
+	dtos := make([]*adminv1.UserAuthToken, len(authTokens))
+	for i, t := range authTokens {
+		var authClientID, authClientDisplayName, representingUserID string
+		var expiresOn *timestamppb.Timestamp
+		if t.AuthClientID != nil {
+			authClientID = *t.AuthClientID
+		}
+		if t.AuthClientDisplayName != nil {
+			authClientDisplayName = *t.AuthClientDisplayName
+		}
+		if t.RepresentingUserID != nil {
+			representingUserID = *t.RepresentingUserID
+		}
+		if t.ExpiresOn != nil {
+			expiresOn = timestamppb.New(*t.ExpiresOn)
+		}
+
+		dtos[i] = &adminv1.UserAuthToken{
+			Id:                    t.ID,
+			DisplayName:           t.DisplayName,
+			AuthClientId:          authClientID,
+			AuthClientDisplayName: authClientDisplayName,
+			RepresentingUserId:    representingUserID,
+			CreatedOn:             timestamppb.New(t.CreatedOn),
+			ExpiresOn:             expiresOn,
+			UsedOn:                timestamppb.New(t.UsedOn),
+		}
+	}
+
+	return &adminv1.ListUserAuthTokensResponse{
+		Tokens:        dtos,
+		NextPageToken: nextToken,
+	}, nil
+}
+
+func (s *Server) IssueUserAuthToken(ctx context.Context, req *adminv1.IssueUserAuthTokenRequest) (*adminv1.IssueUserAuthTokenResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.user_id", req.UserId),
+		attribute.String("args.client_id", req.ClientId),
+		attribute.String("args.display_name", req.DisplayName),
+		attribute.Int64("args.ttl_minutes", req.TtlMinutes),
+		attribute.Bool("args.has_represent_email", req.RepresentEmail != ""),
+	)
+
+	claims := auth.GetClaims(ctx)
+	forceAccess := claims.Superuser(ctx) && req.SuperuserForceAccess
+
+	userID := req.UserId
+	if userID == "current" { // Special alias for the current user
+		if claims.OwnerType() != auth.OwnerTypeUser {
+			return nil, status.Error(codes.Unauthenticated, "not authenticated as a user")
+		}
+		userID = claims.OwnerID()
+	}
+	if userID != claims.OwnerID() && !forceAccess {
```

Comment:
In which case `userID != claims.OwnerID()` will be true as I can see currently all API calls set `req.UserId` as `current`. Thus `userID` will always be same as `claims.OwnerID()` because of line 253. 

Is this to prevent unfaithful access but can only happen if someone knows the userId of a user but isn't user id already protected and thats why we are using special `current` keyword? anyways no harm just curious